s3prl/downstream/runner.py

```diff
@@ -390,3 +410,82 @@ def evaluate(self, split=None, logger=None, global_step=0):
             shutil.rmtree(tempdir)
 
         return [] if type(save_names) is not list else save_names
+
+    def inference(self):
+        filepath = Path(self.args.evaluate_split)
+        assert filepath.is_file()
+        filename = filepath.stem
+
+        if hasattr(self.downstream.model, "load_audio"):
+            wav = self.downstream.model.load_audio(filepath)
+        else:
+            wav, sr = torchaudio.load(str(filepath))
+            assert sr == SAMPLE_RATE
+        wavs = [wav.view(-1).to(self.args.device)]
+
+        for entry in self.all_entries:
+            entry.model.eval()
+
+        with torch.no_grad():
+            features = self.upstream.model(wavs)
+            features = self.featurizer.model(wavs, features)
+            self.downstream.model.inference(features, [filename])
+
+    def push_to_huggingface_hub(self):
+        # Setup auth
+        hf_user = os.environ.get("HF_USERNAME")
+        hf_password = os.environ.get("HF_PASSWORD")
+        huggingface_token = HfApi().login(username=hf_user, password=hf_password)
+        HfFolder.save_token(huggingface_token)
+        print(f"HF Hub user: {hf_user}")
+        
+        # Create repo on the Hub
+        upstream_model = self.args.upstream.replace("/", "__")
+        repo_name = f"superb-s3prl-{upstream_model}-{self.args.downstream}-{str(uuid.uuid4())[:8]}"
+        repo_url = HfApi().create_repo(
+            token=huggingface_token,
+            name=repo_name,
+            organization=hf_user,
+            exist_ok=True,
+            private=True,
+        )
+        print(f"Created Hub repo: {repo_url}")
+
+        # Download repo and copy templates
+        REPO_PATH = self.args.expdir + "/hub_repo/"
+        model_repo = Repository(
+            local_dir=REPO_PATH, clone_from=repo_url, use_auth_token=huggingface_token
+        )
+        TEMPLATES_PATH = f"./downstream/{self.args.downstream}/hf_hub_templates/"
+        shutil.copytree(TEMPLATES_PATH, REPO_PATH, dirs_exist_ok=True)
+
+        # Copy checkpoints, tensorboard logs, and args / configs
+        shutil.copytree(self.args.expdir, REPO_PATH, dirs_exist_ok=True)
```

Comment:
I find that the current file-copying will result in the duplicated `hub_repo` folders. That is, we first create a `Repository` at `hub_repo` under the `expdir`, and then we copy everything inside `expdir` into `hub_repo` including the `hub_repo` itself which already resides in `expdir`. Is this expected? Originally I think we want to copy all the files excluding `hub_repo` itself into `hub_repo` for the upload.

Reply:
Another good catch! I'll fix this as well